packages/auth/src/core/auth/password_policy_impl.ts

```diff
@@ -69,8 +69,10 @@ export class PasswordPolicyImpl implements PasswordPolicyInternal {
       this.enforcementState = 'OFF';
     }
 
+    // Use an empty string if no non-alphanumeric characters are specified in the response.
     this.allowedNonAlphanumericCharacters =
-      response.allowedNonAlphanumericCharacters.join('');
+      response.allowedNonAlphanumericCharacters?.join('') ?? '';
```

Comment:
Let me know if you think this is unreadable. I added a comment to explain what this does.